src/Lecture1.hs

```diff
@@ -104,7 +117,12 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+strSum :: String -> Integer
+strSum str = sum nums
+  where
+    wordsList = words str
+    nums = map closure wordsList
+    closure = (read::String -> Integer)
```

Comment:
As you specify the top-level type signature for this function, the compiler would be able to figure out what type `read` function has in this case due to type inference, so you can skip specifying it in the body:

```suggestion
    closure = read
```